articles/android-rxjava/index.md

```diff
@@ -0,0 +1,127 @@
+Flowable works the same way as an observable but with one extra capability. It is backpressure-aware. Sometimes the observable may emit data at a faster rate than the observer can consume. This can cause the application to crash or get an OOM (Out Of Memory) exception. Due to this, we use the flowable observable, which handles backpressure on behalf of the developer. A common way to get a backpressure issue is when loading a massive list of data from a local database.
+
+##### `single`
+This is another observable which emits data only once, as the name suggests. It does not have the onNext callback since it emits data once. It has the onSuccess instead. It emits the single piece of data in that callback. It is best used when making network calls or retrieving a single entity from a local database.
+
+##### `completable`
+This observable does not emit any data, and so it does not have the `onNext` or `onSuccess` callbacks. We use it to check the completion of a specific action by calling the onCompleted function. It is used for tasks like writing to a local database, uploading an image to a server, etc.
+
+#### Schedulers
+ThThreading is a crucial factor to consider when creating an application. One needs to handle operations on threads carefully. In Android, we have the main thread and UI thread, which do most of the work. It is therefore not advisable to run extra operations like network calls on them. RxJava comes in handy to manage how we deal with threads as we add observers to the application. Schedulers allow us to define where to perform actions and where to receive the data from observables. To specify where an event is to occur, we call the `subscibeOn` method and pass in the thread. We have several options to pass into the subscibeOn method. They are :
+
+##### `newThread`
+We call `Schedulers.newThread()` to specify that the task is to be done on a new thread. This creates a new thread dedicated to that specific task. However, it should be done carefully to prevent having many threads created and running at the same time. That leads to low performance and high CPU usage by the application.
+
+##### `io`
+This creates a new thread that does tasks that do not require a lot of computational power. The observable can also reuse these threads, and if none is available, it creates a new one. We call the `Schedulers.io()` to specify this scheduler to be used. It handles tasks like making network calls and file system tasks.
+
+##### `computation`
+Computation works in the same way as the io scheduler, only that we use it for CPU intensive work. However, it creates a limited number of threads according to the number of cores in the Android device. We use it for tasks like reading local databases, and should be used carefully. Due to the limited number of threads, if a job finds all the threads in use, it has to wait for them to finish the current tasks. Observables use `Schedulers.computation()` to run tasks using this scheduler.
```

Comment:
"Computation works similar to the io scheduler" flows better.

Reply:
done...